src/main/java/net/javadiscord/javabot/systems/user_commands/format_code/FormatCodeCommand.java

```diff
@@ -0,0 +1,92 @@
+package net.javadiscord.javabot.systems.user_commands.format_code;
+
+import com.dynxsty.dih4jda.interactions.commands.SlashCommand;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.javadiscord.javabot.util.InteractionUtils;
+import net.javadiscord.javabot.util.Responses;
+import net.javadiscord.javabot.util.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * <h3>This class represents the /format-code command.</h3>
+ */
+public class FormatCodeCommand extends SlashCommand {
+	/**
+	 * The constructor of this class, which sets the corresponding {@link net.dv8tion.jda.api.interactions.commands.build.SlashCommandData}.
+	 */
+	public FormatCodeCommand() {
+		setSlashCommandData(Commands.slash("format-code", "Format unformatted code from a message")
+				.setGuildOnly(true)
+				.addOptions(
+						new OptionData(OptionType.STRING, "message-id", "Message to be formatted, last message used if left blank.", false),
+						new OptionData(OptionType.STRING, "format", "The language used to format the code, defaults to Java if left blank.", false)
+								.addChoice("C", "c")
+								.addChoice("C#", "csharp")
+								.addChoice("C++", "cpp")
+								.addChoice("CSS", "css")
+								.addChoice("D", "d")
+								.addChoice("Go", "go")
+								.addChoice("HTML", "html")
+								.addChoice("Java", "java")
+								.addChoice("JavaScript", "js")
+								.addChoice("Kotlin", "kotlin")
+								.addChoice("PHP", "php")
+								.addChoice("Python", "python")
+								.addChoice("Ruby", "ruby")
+								.addChoice("Rust", "rust")
+								.addChoice("SQL", "sql")
+								.addChoice("Swift", "swift")
+								.addChoice("TypeScript", "typescript")
+								.addChoice("XML", "xml")
+				)
+		);
+	}
+
+	protected static ActionRow buildActionRow(Message target) {
```

Comment:
Why is this `protected static`?

Reply:
So that it can be used in FormatCodeMessageContext.java

Reply:
`FormatCodeMessageContext extends FormatCodeCommand` and `FormatCodeCommand extends SlashCommand` resulting in `FormatCodeCommand` extending `SlashCommand`. Does this not result in the same `SlashCommand` being registered multiple times? Wouldn't it make more sense to create a single class for both?